ACM-26604 Observability alertmanager secret rename

jacobbaungard opened the pull request:
Fixes: https://issues.redhat.com/browse/ACM-26604

jacobbaungard committed:
ACM-26604 Observability alertmanager secret rename

Fixes: https://issues.redhat.com/browse/ACM-26604

Signed-off-by: Jacob Baungard Hansen <jacobbaungard@redhat.com>

jacobbaungard committed:
Mention both secrets changes

Signed-off-by: Jacob Baungard Hansen <jacobbaungard@redhat.com>

dockerymick commented on release_notes/known_issues_observability.adoc:
```diff
@@ -7,6 +7,12 @@ For your {ocp} cluster, see link:https://docs.redhat.com/en/documentation/opensh
+In {acm-short} 2.15 the name of the `observability-alertmanager-accessor`, and `hub-alertmanager-router-ca` secrets were changed to include a postfix of the hub cluster name, for example: `observability-alertmanager-accessor-hub1-region1`. On systems where the automatic handling for alert-forwarding config has been disabled with the annotation: `mco-disable-alerting: true` on the `MultiClusterObs ...
+
+For platform alerts the `cluster-monitoring-config` configmap in namespace `openshift-monitoring` should be updated and for user-workload alerts the configmap `user-workload-monitoring-config` in namespace: ` openshift-user-workload-monitoring` should be updated. The exact secret names can be found by viewing the secrets in the mentioned namespaces.
+
```
I suggest to update the title to reduce the length of it and update the anchor to align with the new title. For the description, we should reiterate the problem and then explain why it's happening.

Extra line space is required so that the topic renders correctly.

the changes in my suggestion help to have a active tone. Should the user update the `mco-disable-alerting` to `false`? We just tell the user that they have to change the configuration.

Let me know your thoughts @jacobbaungard 

```suggestion
[#alert-forward-fail-after-upgrade]
== Manually configured alert forwarding fails after upgrading to {acm-short} 2.15

After you upgrade to version 2.15, alert forwarding fails. In {acm-short} 2.15, the name of the `observability-alertmanager-accessor` and name of the `hub-alertmanager-router-ca` secrets were changed to include a postfix of the hub cluster name. See the following example of how the name is displayed: `observability-alertmanager-accessor-hub1-region1`. 

To fix this issue, you must update the configurations where you disabled automatic alert forwarding with the annotation, `mco-disable-alerting: true` in the `MultiClusterObservability` custom resource, and where you manually configured alert forwarding to forward alerts to your hub cluster. 

For platform alerts, update the `cluster-monitoring-config` config map in the `openshift-monitoring` namespace. For user workload alerts, update the `user-workload-monitoring-config` config map  in the `openshift-user-workload-monitoring` namespace. View the secret names that are in the `openshift-monitoring` and `openshift-user-workload-monitoring` namespaces.

```

jacobbaungard committed:
apply doc review suggestions

Signed-off-by: Jacob Baungard Hansen <jacobbaungard@redhat.com>

dockerymick commented on release_notes/known_issues_observability.adoc:
```diff
@@ -7,6 +7,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/en/documentation/opensh
+[#alert-forward-fail-after-upgrade]
+== Manually configured alert forwarding fails after upgrading to {acm-short} 2.15
+
+After you upgrade to version 2.15, alert forwarding fails, if you have manually configured alert forwarding. In {acm-short} 2.15, the name of the `observability-alertmanager-accessor` and `hub-alertmanager-router-ca` secrets were changed to include a postfix of the hub cluster name. See the following example of how the name is displayed: `observability-alertmanager-accessor-hub1-region1`. 
```
What do you think about us specifying what it means to manually configure alert forwarding? Does this problem happen if the user configures through the CLI or the console? Does it happen despite how the config map was updated? 

For example, "If you manually configured alert manager by making changes directly to your `MultiClusterObservability` custom resource and upgrade to version 2.15, alert forwarding fails. Alert forwarding fails after you upgrade because the name of the `observability-alertmanager-accessor` and `hub-alertmanager-router-ca` secrets were changed to include a postfix of the hub cluster name."

```suggestion
If you manually configured alert manager by making changes directly to your `MultiClusterObservability` custom resource and upgrade to version 2.15, alert forwarding fails. Alert forwarding fails after you upgrade because the name of the `observability-alertmanager-accessor` and `hub-alertmanager-router-ca` secrets were changed to include a postfix of the hub cluster name. 
```

jacobbaungard replied:
There are multiple ways to do it manually, people do it using gitops, using policies, or completely manually by changing the configmap directly. I don't think we need to specify the options for how to do it here.

The only thing is that `mco-disable-alerting: true` should be set, and that is mentioned in the next paragraph.

jacobbaungard committed:
another try

Signed-off-by: Jacob Baungard Hansen <jacobbaungard@redhat.com>

dockerymick committed:
Rewrite after dev review

Clarified instructions for updating alert forwarding configuration after upgrade.

jc-berger commented on release_notes/known_issues_observability.adoc:
```diff
@@ -7,6 +7,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/en/documentation/opensh
 For more about deprecations and removals, see xref:../release_notes/acm_deprecate_remove.adoc#deprecations-removals-acm[Deprecations and removals for {acm-short}]. 
 
+[#alert-forward-fail-after-upgrade]
+== Manually configured alert forwarding fails after upgrading to {acm-short} 2.15
```
kind of a long title. Can you state the version in the intro sentence to shorten the title? 
```suggestion
== Manually configured alert forwarding fails
```

jc-berger commented on release_notes/known_issues_observability.adoc:
```diff
@@ -7,6 +7,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/en/documentation/opensh
 
 For more about deprecations and removals, see xref:../release_notes/acm_deprecate_remove.adoc#deprecations-removals-acm[Deprecations and removals for {acm-short}]. 
 
+[#alert-forward-fail-after-upgrade]
```
I know changing anchors can be a pain, but this a little long too. what do you think of shortening it? 
```suggestion
[#alert-forward-fails]
```

jc-berger commented on release_notes/known_issues_observability.adoc:
```diff
@@ -7,6 +7,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/en/documentation/opensh
+[#alert-forward-fail-after-upgrade]
+== Manually configured alert forwarding fails after upgrading to {acm-short} 2.15
+
+If you disabled automatic configuration for alert forwarding by setting `mco-disable-alerting: true` in the `MultiClusterObservability` custom resource, and manually configured alert forwarding, alert forwarding fails. Alert forwarding fails after you upgrade because the name of the `observability-alertmanager-accessor` and `hub-alertmanager-router-ca` secrets were changed to include a postfix of ...
```
```suggestion
When you upgrade to {acm-short} {product-version} alert forwarding fails if you disabled automatic configuration by setting `mco-disable-alerting: true` in the `MultiClusterObservability` custom resource, and if you manually configured alert forwarding. The failure happens after you upgrade because the name of the `observability-alertmanager-accessor` and `hub-alertmanager-router-ca` secrets were changed to include a postfix of the hub cluster name.
```

jacobbaungard replied:
This also sounds a bit better to me, if you are also ok with that @dockerymick ?

jc-berger commented on release_notes/known_issues_observability.adoc:
```diff
@@ -7,6 +7,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/en/documentation/opensh
+
+If you disabled automatic configuration for alert forwarding by setting `mco-disable-alerting: true` in the `MultiClusterObservability` custom resource, and manually configured alert forwarding, alert forwarding fails. Alert forwarding fails after you upgrade because the name of the `observability-alertmanager-accessor` and `hub-alertmanager-router-ca` secrets were changed to include a postfix of ...
+
+To fix this issue, you must update your alert forwarding configuration that you manually configured to use the new names of the secrets.
```
```suggestion
To fix this issue, update your alert forwarding configuration that you manually configured to use the new names of the secrets.
```

jc-berger commented on release_notes/known_issues_observability.adoc:
```diff
@@ -7,6 +7,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/en/documentation/opensh
+
+To fix this issue, you must update your alert forwarding configuration that you manually configured to use the new names of the secrets.
+
+For platform alerts, update secret name for the `cluster-monitoring-config` config map in the `openshift-monitoring` namespace. For user workload alerts, update the secret name of the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` namespace. Find the new secret names in the `openshift-monitoring` and `openshift-user-workload-monitoring` namespaces.
```
```suggestion
For platform alerts, update the secret name for the `cluster-monitoring-config` config map in the `openshift-monitoring` namespace. For user workload alerts, update the secret name of the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` namespace. Find the new secret names in the `openshift-monitoring` and `openshift-user-workload-monitoring` namespaces.
```

jc-berger commented on release_notes/known_issues_observability.adoc:
```diff
@@ -7,6 +7,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/en/documentation/opensh
+To fix this issue, you must update your alert forwarding configuration that you manually configured to use the new names of the secrets.
+
+For platform alerts, update secret name for the `cluster-monitoring-config` config map in the `openshift-monitoring` namespace. For user workload alerts, update the secret name of the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` namespace. Find the new secret names in the `openshift-monitoring` and `openshift-user-workload-monitoring` namespaces.
+
```
there should be the link to the bug issue or dev issue here. Looking at your jira, I could not find one. But I used the jira assigned to this PR. Perhaps that suffices, and if not, you at least have the template to change the link: 
```suggestion
link:https://issues.redhat.com/browse/ACM-26604[ACM-26604]
```

dockerymick committed:
Apply suggestions from code review

Co-authored-by: jc-berger <70717303+jc-berger@users.noreply.github.com>

dockerymick reviewed (approved):
Thank you so much @jc-berger for your thorough review!

dockerymick committed:
Update known_issues_observability.adoc

jacobbaungard committed:
Clarify the changes go _in_ the configmap

Signed-off-by: Jacob Baungard Hansen <jacobbaungard@redhat.com>

dockerymick committed:
Update release_notes/known_issues_observability.adoc

Co-authored-by: jc-berger <70717303+jc-berger@users.noreply.github.com>

jc-berger reviewed (approved):
/lgtm
